com.unity.shadergraph/Editor/Drawing/Inspector/PropertyDrawers/ShaderInputPropertyDrawer.cs

```diff
@@ -94,13 +94,55 @@ public ShaderInputPropertyDrawer()
 
         public Action inspectorUpdateDelegate { get; set; }
 
+        private void AddDeprecatedVisualElementsIfNeeded(PropertySheet propertySheet, AbstractShaderProperty property)
+        {
+            if (property.version < property.latestVersion || (ShaderGraphPreferences.allowDeprecatedBehaviors && property.latestVersion > 0))
+            {
+                var typeString = property.propertyType.ToString();
+                HelpBoxRow help;
+                if (ShaderGraphPreferences.allowDeprecatedBehaviors && property.latestVersion > 0)
+                {
+                    help = new HelpBoxRow(MessageType.Info);
+                    help.Add(new Label("OVERRIDE: Hover for info")
+                    {
+                        tooltip = $"The {typeString} Property has new updates. This version maintains the old behavior. See the {typeString} Property documentation for more information."
+                    });
+                    int[] entries = Enumerable.Range(0, property.latestVersion + 1).ToArray();
+                    help.Add(new IMGUIContainer(() =>
+                    {
+                        EditorGUI.BeginChangeCheck();
+                        int newVersion = EditorGUILayout.IntPopup(property.version, entries.Select(i => i.ToString()).ToArray(), entries);
+                        if(EditorGUI.EndChangeCheck())
+                        {
+                            property.ChangeVersion(newVersion);
+                        }
+                    }));
+                }
+                else
+                {
+                    help = new HelpBoxRow(MessageType.Warning);
+                    var label = new Label("DEPRECATED: Hover for info")
+                    {
+                        tooltip = $"The {typeString} Property has new updates. This version maintains the old behavior. If you update a {typeString} Property, you can use Undo to change it back. See the {typeString} Property documentation for more information."
+                    };
+                    help.Add(label);
+                    help.contentContainer.Add(new Button(() => property.ChangeVersion(property.latestVersion)) { text = "Update" } );
+                }
+                propertySheet.Add(help);
+            }
+        }
+
         public VisualElement DrawProperty(
             PropertyInfo propertyInfo,
             object actualObject,
             InspectableAttribute attribute)
         {
             var propertySheet = new PropertySheet();
             shaderInput = actualObject as ShaderInput;
+            if (shaderInput is AbstractShaderProperty property)
+            {
+                AddDeprecatedVisualElementsIfNeeded(propertySheet, property);
+            }
```

Comment:
probably can add this to the BuildPropertyField function call and use insert(0) instead of add